File: Version0.2/Julia/Mesh/squaremesh.jl

function TriSquareMesh(
    m::Int,
    n::Int = m,
)::Tuple{Array{Float64,2},Array{Int,2}}

    p = [[x; y] for x = 0.0:1.0/Float64(m):1.0, y = 0.0:1.0/Float64(n):1.0][:]
    t = [
        [
            [i + (j - 1) * (m + 1) i + (j - 1) * (m + 1) + 1 i + j * (m + 1)],
            [i + (j - 1) * (m + 1) + 1 i + j * (m + 1) + 1 i + j * (m + 1)],
        ] for i = 1:m, j = 1:n
    ]

    return hcat(p...), hcat(hcat(t'...)...)
end

function QuadSquareMesh(
    m::Int,
    n::Int = m,
)::Tuple{Array{Float64,2},Array{Int,2}}

    p = [[x; y] for x = 0.0:1.0/Float64(m):1.0, y = 0.0:1.0/Float64(n):1.0][:]
    t = [ [i + (j - 1) * (m + 1)    i + (j - 1) * (m + 1) + 1  i + j * (m + 1) + 1   i + j * (m + 1)]
          for j = 1:m, i = 1:n ]

    return hcat(p...), hcat(t'...)
end

function SquareMesh(
    m::Int,
    n::Int,
    elemtype::Int,
)::Tuple{Array{Float64,2},Array{Int,2}}

    if (elemtype==0)
        p, t = TriSquareMesh(m,n)
    else
        p, t = QuadSquareMesh(m,n)
    end

    return p, t
end

File: Version0.1/Python/Mesh/squaremesh.py
import numpy as np

def TriSquareMesh(m, n):

    p = [[x/m, y/n] for y in np.arange(0,n+1) for x in np.arange(0,m+1)];
    p = np.array(p,order='F');

    t = [ [ [i + (j - 1) * (m + 1), i + (j - 1) * (m + 1) + 1, i + j * (m + 1)], [i + (j - 1) * (m + 1) + 1, i + j * (m + 1) + 1, i + j * (m + 1)]] for i in range (1,m+1) for j in range (1,n+1) ]
    t = np.array(t);
    t = np.reshape(t,(2*m*n,3));

    return p, t

def QuadSquareMesh(m,n):

    p = [[x/m, y/n] for y in np.arange(0,n+1) for x in np.arange(0,m+1)];
    p = np.array(p,order='F');

    t = [ [i + (j - 1) * (m + 1), i + (j - 1) * (m + 1) + 1,  i + j * (m + 1) + 1,   i + j * (m + 1)] for j in range (1,m+1) for i in range (1,n+1)];
    t = np.array(t);

    return p, t

def SquareMesh(m, n, elemtype):

    if (elemtype==0):
        p, t = TriSquareMesh(m,n)[0:2]
    else:
        p, t = QuadSquareMesh(m,n)[0:2]

    p = p.transpose();
    t = t.transpose() - 1;

    return p, t
